tests/charts/api_tests.py

```diff
@@ -634,19 +636,43 @@ def test_get_charts_no_data_access(self):
         data = json.loads(rv.data.decode("utf-8"))
         self.assertEqual(data["count"], 0)
 
-    def test_chart_data(self):
+    def test_chart_data_simple(self):
         """
         Query API: Test chart data query
         """
         self.login(username="admin")
         table = self.get_table_by_name("birth_names")
-        payload = get_query_context(table.name, table.id, table.type)
-        uri = "api/v1/chart/data"
-        rv = self.post_assert_metric(uri, payload, "data")
+        request_payload = get_query_context(table.name, table.id, table.type)
+        rv = self.post_assert_metric(CHART_DATA_URI, request_payload, "data")
         self.assertEqual(rv.status_code, 200)
         data = json.loads(rv.data.decode("utf-8"))
         self.assertEqual(data["result"][0]["rowcount"], 100)
 
+    def test_chart_data_limit_offset(self):
+        """
+        Query API: Test chart data query with limit and offset
+        """
+        self.login(username="admin")
+        table = self.get_table_by_name("birth_names")
+        request_payload = get_query_context(table.name, table.id, table.type)
+        request_payload["queries"][0]["row_limit"] = 5
```

Comment:
Would be nice to a couple of simple tests with invalid `row_limit` and `row_offset`. negative values, bigger then `SAMPLES_ROW_LIMIT`

Reply:
Added schema validation tests (`tests/chart/schema_tests.py`) for validation of `row_limit ` and `row_offset` and e2e tests (`tests/chart/api_tests.py`) to ensure regular and sample row limits produce correct row counts.